### PR TITLE
Python 3.11 with a patch to current pickler.

### PR DESCRIPTION
**Pull request by tvalentyn**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by tvalentyn** [comment]
Run Seed Job

**Comment by tvalentyn** [comment]
Run Seed Job

File: sdks/python/container/base_image_requirements_manual.txt [hunk 1]
```diff
@@ -32,11 +32,11 @@ cython<1
 # some versions of libraries that launch Beam pipelines, like tensorflow-transform.
 # Leaving 'future' in our containers for now prevent breaking tft users.
 future
-guppy3  # Memory profiler
+guppy3;python_version<"3.11"  # Memory profiler
```
Comment (tvalentyn) [inline sdks/python/container/base_image_requirements_manual.txt:35]: @AnandInguva can we link bugs here?

**Comment by tvalentyn** [comment]
Run Seed Job

**Comment by tvalentyn** [comment]
Seed job: https://ci-beam.apache.org/job/beam_SeedJob/11365/

File: sdks/python/container/base_image_requirements_manual.txt [hunk 2]
```diff
@@ -32,11 +32,11 @@ cython<1
...
 mmh3  # Optimizes execution of some Beam codepaths. TODO: Make it Beam's dependency.
 nltk  # Commonly used for natural language processing.
 nose==1.3.7  # For Dataflow internal testing. TODO: remove this.
-python-snappy  # Optimizes execution of some Beam codepaths.
+python-snappy;python_version<"3.11"  # Optimizes execution of some Beam codepaths.
```
Comment (AnandInguva) [inline sdks/python/container/base_image_requirements_manual.txt:39]:
```suggestion
# TODO: Uncomment python version once python-snappy supports it.
# https://github.com/apache/beam/issues/25985
python-snappy;python_version<"3.11"  # Optimizes execution of some Beam codepaths.
```

File: sdks/python/container/base_image_requirements_manual.txt (hunk 1 above)
Comment (AnandInguva) [inline sdks/python/container/base_image_requirements_manual.txt:35]:
```suggestion
# TODO: Uncomment python version once guppy supports it.
# https://github.com/apache/beam/issues/25984
guppy3;python_version<"3.11"  # Memory profiler
```

**Comment by tvalentyn** [comment]
Run Python Dataflow V2 ValidatesRunner

**Comment by AnandInguva** [comment]
Run Python 3.11 PostCommit

**Comment by AnandInguva** [comment]
Run Python 3.11 PostCommit

**Comment by tvalentyn** [comment]
Noting that tests have passed, but status have not propagated.
Will merge these changes in https://github.com/apache/beam/pull/26121

**Pull request closed without merging**